lib/client/mfa/cli.go

```diff
@@ -101,40 +107,71 @@ func (c *CLIPrompt) Run(ctx context.Context, chal *proto.MFAAuthenticateChalleng
 
 	promptOTP := chal.TOTP != nil
 	promptWebauthn := chal.WebauthnChallenge != nil
+	promptSSO := false // TODO(Joerger): check for SSO challenge once added in separate PR.
 
 	// No prompt to run, no-op.
-	if !promptOTP && !promptWebauthn {
+	if !promptOTP && !promptWebauthn && !promptSSO {
 		return &proto.MFAAuthenticateResponse{}, nil
 	}
 
+	var availableMethods []string
+	if promptWebauthn {
+		availableMethods = append(availableMethods, cliMFATypeWebauthn)
+	}
+	if promptSSO {
+		availableMethods = append(availableMethods, cliMFATypeSSO)
+	}
+	if promptOTP {
+		availableMethods = append(availableMethods, cliMFATypeOTP)
+	}
+
 	// Check off unsupported methods.
 	if promptWebauthn && !c.WebauthnSupported {
 		promptWebauthn = false
 		slog.DebugContext(ctx, "hardware device MFA not supported by your platform")
-		if !promptOTP {
-			return nil, trace.BadParameter("hardware device MFA not supported by your platform, please register an OTP device")
-		}
 	}
 
 	// Prefer whatever method is requested by the client.
+	var chosenMethods []string
 	switch {
+	case c.PreferSSO && promptSSO:
+		chosenMethods = []string{cliMFATypeSSO}
+		promptWebauthn, promptOTP = false, false
 	case c.PreferOTP && promptOTP:
-		promptWebauthn = false
+		chosenMethods = []string{cliMFATypeOTP}
+		promptWebauthn, promptSSO = false, false
 	}
 
 	// Use stronger auth methods if hijack is not allowed.
-	if !c.AllowStdinHijack && promptWebauthn {
+	if !c.AllowStdinHijack && (promptWebauthn || promptSSO) {
 		promptOTP = false
 	}
 
-	// If we have multiple viable options, prefer Webauthn > OTP.
+	// If we have multiple viable options, prefer Webauthn > SSO > OTP.
 	switch {
 	case promptWebauthn:
+		chosenMethods = []string{cliMFATypeWebauthn}
+		promptSSO = false
+
 		// If a specific webauthn attachment was requested, skip OTP.
 		// Otherwise, allow dual prompt with OTP.
 		if c.AuthenticatorAttachment != wancli.AttachmentAuto {
 			promptOTP = false
+		} else if promptOTP {
+			chosenMethods = append(chosenMethods, cliMFATypeOTP)
 		}
```

Comment:
Maybe?

```suggestion

		// If a specific webauthn attachment was requested, skip OTP.
		// Otherwise, allow dual prompt with OTP.
		promptOTP := promptOTP && c.AuthenticatorAttachment == wancli.AttachmentAuto

		if promptOTP {
			chosenMethods = append(chosenMethods, cliMFATypeOTP)
		}
```